test/e2e/storage/testsuites/base.go

```diff
@@ -199,7 +199,7 @@ func CreateVolumeResource(driver TestDriver, config *PerTestConfig, pattern test
 		framework.Logf("Creating resource for inline volume")
 		if iDriver, ok := driver.(InlineVolumeTestDriver); ok {
 			r.VolSource = iDriver.GetVolumeSource(false, pattern.FsType, r.Volume)
-			r.VolType = dInfo.Name
+			r.VolType = shortenCSIDriverName(dInfo.Name)
```

Comment:
My understanding is that dInfo.Name comes from the user-provided [external config file](https://github.com/kubernetes/kubernetes/blame/master/test/e2e/storage/external/README.md#L18), and this doesn't actually have to equal the full csi driver name, it just needs to be a descriptive name. Are you seeing otherwise?

If it doesn't actually have to match the driver name, can we just document the name length limit?

Reply:
My understanding is that it has to be the full CSI driver name: The name is first [declared as the provisioner](https://github.com/kubernetes/kubernetes/blob/78c56e6b4cf963e5d773908ef9780345d216188b/test/e2e/storage/external/external.go#L263) and then [fed into the StorageClass value](https://github.com/kubernetes/kubernetes/blob/78c56e6b4cf963e5d773908ef9780345d216188b/test/e2e/storage/testsuites/driveroperations.go#L80) we create.

I also just ran a quick test and selected a different name, but that makes tests hang/fail with

```
waiting for a volume to be created, either by external provisioner "foobar-driver" or manually created by system administrator
```

I think the tests need some way to reference the provisioner, and at least with the name-based approach the name has to match that.

Please let me know if I misunderstood.

Reply:
Looking at the external CSI hostpath driver, it seems that it also [defines the full name](https://github.com/kubernetes-csi/csi-driver-host-path/blob/5a4616d5fa4951366a998822c718d939f4f5275a/.prow.sh#L11) and [injects it into the YAML configuration file accordingly](https://github.com/kubernetes-csi/csi-driver-host-path/blob/5a4616d5fa4951366a998822c718d939f4f5275a/release-tools/prow.sh#L815).

I also see [the documentation getting updated to use what looks like the full name](https://github.com/kubernetes/kubernetes/pull/86077/files#diff-5f7a61af7716b17d32155c03c6133f2cR19).

Reply:
Ok thanks for pointing that out. I'm actually trying to search for which tests are using `r.VolType`, and I only found the one here: https://github.com/kubernetes/kubernetes/blob/a6f41a46a54c47e4225adbcca251ce8204f9bb1c/test/e2e/storage/testsuites/volumes.go#L211

I'm actually wondering if we can remove this `r.VolType` altogether, and use a different naming scheme for that test.

Reply:
There's another occurrence in https://github.com/kubernetes/kubernetes/blob/eef4c00ae93bd51125c215918a7b1c632d298610/test/e2e/storage/testsuites/subpath.go#L503

However, it's not just one or two tests that are affected due to the dynamic nature of how storage tests are generated. Below is an example output of our CSI tests running _without_ the patch this PR proposes, showing 16 failures seemingly all of them failing due to the length problem:

```
Summarizing 16 Failures:

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should fail if subpath file is outside the volume [Slow][LinuxOnly]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support creating multiple subpath from same volumes [Slow]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support restarting containers using directory as subpath [Slow]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should fail if subpath directory is outside the volume [Slow]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support file as subpath [LinuxOnly]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support existing single file [LinuxOnly]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support existing directories when readOnly specified in the volumeSource

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support readOnly file specified in the volumeMount [LinuxOnly]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support restarting containers using file as subpath [Slow][LinuxOnly]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should be able to unmount after the subpath directory is deleted

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support non-existent path

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support existing directory

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should fail if subpath with backstepping is outside the volume [Slow]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should support readOnly directory specified in the volumeMount

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should fail if non-existent subpath is outside the volume [Slow][LinuxOnly]

[Fail] External Storage [Driver: dobs.csi.digitalocean.com-dev] [Testpattern: Dynamic PV (default fs)] subPath [It] should verify container cannot write to subpath readonly volumes [Slow]

Ran 27 of 5062 Specs in 394.036 seconds
FAIL! -- 11 Passed | 16 Failed | 0 Pending | 5035 Skipped
```

This is only to show the extent of the tests that are affected; it's not to say we possibly can't address this problem by fixing the two occurrences of the `generateSuffixForPodName` function call (or its implementation).

Reply:
I'm wondering if we need to include the volume type / CSI driver name in the pod name at all -- couldn't we just omit it?

Reply:
I tracked git history down where the volume type was first introduced for the pod name suffix: it looks like the [addition of the e2e subpath tests](https://github.com/kubernetes/kubernetes/commit/ba818d8dffd0f8c76471e05d931b20df6569fa2a#diff-c803c0d0e10a93e495e668a6722cd77aR201-R202) brought it in.

@jsafrane I see you were the author back then. Maybe you have context around whether the volume type is strictly necessary to be part of the suffix name.

Reply:
To ease the discussion, I just pushed a commit that omits the volume type from the pod name suffix. The tests still ran successfully for my CSI driver with that change.

Reply:
> Maybe you have context around whether the volume type is strictly necessary to be part of the suffix name.

No, I just wanted something that's easy to read in e2e logs. The pod names can be virtually anything.

Reply:
Yeah I like the 2nd commit better :-)  Let's also just remove `r.VolType` altogether?

Reply:
Removed `r.VolType` and made some necessary adjustments. (Parts of `subpath.go` were still referencing it.)

I tried to reconcile both desires to have pod names readable in logs and long driver names not break tests by only including the actual volume type (i.e., excluding the driver name) in pod name suffices. I feel that's a good compromise between the two goals.

Let me know what you think.